src/main/java/com/icthh/xm/ms/otp/service/impl/OneTimePasswordServiceImpl.java

```diff
@@ -96,7 +96,12 @@ public OneTimePasswordDto generate(OneTimePasswordDto oneTimePasswordDto) {
         oneTimePasswordRepository.saveAndFlush(otp);
 
         communicationService.sendOneTimePassword(message, otp.getReceiver(), otpType.getOtpSenderId());
-        return oneTimePasswordMapper.toDto(otp);
+
+        OneTimePasswordDto otpResult = oneTimePasswordMapper.toDto(otp);
+
+        otpResult.setLangKey(oneTimePasswordDto.getLangKey());
```

Comment:
I suggest to move this logic to mapper:
com.icthh.xm.ms.otp.service.mapper.OneTimePasswordMapperImpl#toDto(com.icthh.xm.ms.otp.domain.OneTimePassword)